### DISCUSSION TITLE
How to set state from value being watched

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#5316] [Q&A] [answered]
I am using useWatch to show the form total in real time to the user. But when the user submits I want to send this value but it isn't actually a form input. It doesn't like it when I try to save this value to state using useState so not sure what to do?

```
  const PriceTotal = ({ control }) => {
    const value = useWatch({
      control,
      name: `items`,
      defaultValue: {},
    });

    let invoiceTotal;
    if (value && value.length) {
      invoiceTotal = value.reduce((prev, cur) => prev + cur.price * cur.qty, 0);
    } else {
      console.log("no values yet");
    }
    return <span>{invoiceTotal}</span>;
  };
```

**Comment by bluebill1049** [comment]
Register the value as a hidden input?

**Comment by ghost** [comment]
Yeah, I thought of that but was hoping I could rather have it state. If not possible then I guess hidden field is my next best bet.

**Comment by thanh-nguyen-95** [comment, marked as answer]
You can merge it the the `formValues` when submit the form like so:

```jsx
const values = useWatch({ name: 'items', defaultValue: {}, control })

const invoiceTotal = useMemo(() => { /* calculate total here */ }, [values])

const onSubmit = useCallback((formValues) => {
  const formValuesWithInvoiceTotal = { ...formValues, invoiceTotal }
  /* do something here */
}, [invoiceTotal])

return (
  <form onSubmit={handleSubmit(onSubmit)}></form>
)
```

If you don't like merging, use a hidden input with the value of `invoiceTotal`

**Reply by bluebill1049** [reply]
LGTM.

**Reply by thanh-nguyen-95** [reply]
Lol that `LGTM.` somehow remind me of this video https://www.youtube.com/watch?v=rR4n-0KYeKQ

**Reply by ghost** [reply]
Nice one, thanks for this. I will certainly give it a go.

**Reply by bluebill1049** [reply]
Lol, i watched that video before as well.
